src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -551,47 +486,44 @@ void probe(const RadixContainer<ProbeColumnType>& probe_radix_container,
     Therefore, inputs for one partition should be located on the same NUMA node,
     and the job that probes that partition should also be on that NUMA node.
   */
-  for (size_t current_partition_id = 0; current_partition_id < probe_radix_container.partition_offsets.size();
-       ++current_partition_id) {
-    const auto partition_begin =
-        current_partition_id == 0 ? 0 : probe_radix_container.partition_offsets[current_partition_id - 1];
-    const auto partition_end = probe_radix_container.partition_offsets[current_partition_id];  // make end non-inclusive
 
+  for (size_t partition_idx = 0; partition_idx < probe_radix_container.size(); ++partition_idx) {
     // Skip empty partitions to avoid empty output chunks
-    if (partition_begin == partition_end) {
+    if (probe_radix_container[partition_idx].elements.empty()) {
       continue;
     }
 
-    jobs.emplace_back(std::make_shared<JobTask>([&, partition_begin, partition_end, current_partition_id]() {
-      // Get information from work queue
-      auto& partition = static_cast<Partition<ProbeColumnType>&>(*probe_radix_container.elements);
+    jobs.emplace_back(std::make_shared<JobTask>([&, partition_idx]() {
+      const auto& partition = probe_radix_container[partition_idx];
+      const auto& elements = partition.elements;
+      const auto& null_values = partition.null_values;
+
       PosList pos_list_build_side_local;
-      PosList pos_list_probe_local;
+      PosList pos_list_probe_side_local;
 
       if constexpr (keep_null_values) {
-        DebugAssert(
-            probe_radix_container.null_value_bitvector->size() == probe_radix_container.elements->size(),
-            "Hash join probe called with NULL consideration but inputs do not store any NULL value information");
+        Assert(elements.size() == null_values.size(),
+               "Hash join probe called with NULL consideration but inputs do not store any NULL value information");
       }
 
-      if (hash_tables[current_partition_id]) {
-        const auto& hash_table = hash_tables.at(current_partition_id).value();
+      const auto hash_table_idx = hash_tables.size() > 1 ? partition_idx : 0;
+      if (!hash_tables.empty() && hash_tables.at(hash_table_idx)) {
+        const auto& hash_table = *hash_tables[hash_table_idx];
 
         // Accessors are not thread-safe, so we create one evaluator per job
```

Comment:
Accessors?

Reply:
Changed to "The MultiPredicateJoinEvaluator use accessors internally. Those are not thread-safe, so we create one
evaluator per job."